src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -2342,86 +2341,90 @@ int wm_vuldet_check_agent_vulnerabilities(wm_vuldet_t *vuldet) {
     }
 
     // Iterate agents to look for vulnerabilities
-    for (i = 0, agents_it = vuldet->scan_agents;; i++) {
-        scan_ctx_t scan_ctx = {0};
-        scan_ctx.agent_id = atoi(agents_it->agent_id);
-        scan_ctx.agent_name = agents_it->agent_name;
-        scan_ctx.agent_ip = agents_it->agent_ip;
-        mtinfo(WM_VULNDETECTOR_LOGTAG, VU_START_AG_AN, scan_ctx.agent_id);
-        time_t start = time(NULL);
-
-        // Check if there are available vulnerabilities for this agent
-        if (agents_it->dist != FEED_WIN && agents_it->dist != FEED_MAC) {
-            result = wm_vuldet_db_empty(db, agents_it->dist_ver);
-            if (result == 0) {
-                // There is no data in the VULNERABILITIES table for this agent
-                // It has to be skipped instead of being scanned against the NVD to avoid false positives
-                mtwarn(WM_VULNDETECTOR_LOGTAG, VU_OVAL_UNAVAILABLE_DATA, scan_ctx.agent_id);
-                if (agents_it->next) {
-                    agents_it = agents_it->next;
+    do {
+        pending_agents = false;
+        for (agents_it = vuldet->scan_agents; agents_it; agents_it = agents_it->next) {
+            scan_ctx_t scan_ctx = {0};
+            scan_ctx.agent_id = atoi(agents_it->agent_id);
+            scan_ctx.agent_name = agents_it->agent_name;
+            scan_ctx.agent_ip = agents_it->agent_ip;
+            mtinfo(WM_VULNDETECTOR_LOGTAG, VU_START_AG_AN, scan_ctx.agent_id);
+            time_t start = time(NULL);
+
+            if (agents_it->scanned || (!agents_it->scanned && agents_it->attempts == 0)) continue;
```

Comment:
I think we can simplify this by eliminating agents_it->scanned, and use a unique agents_it->pending_attempts that will be decreased after each failure scanand set to 0 after a successfull scan.

Reply:
You are right! Fixed!